contracts/CompoundAssetProxy.sol

```diff
@@ -0,0 +1,143 @@
+        // Return the amount of shares the user has produced and the amount of underlying used for it.
+        return (shares, amount);
+    }
+
+    /// @notice Withdraw the number of shares
+    /// @param _shares The number of shares to withdraw
+    /// @param _destination The address to send the output funds
+    // @param _underlyingPerShare The possibly precomputed underlying per share
+    /// @return Amount of funds freed by doing a withdraw
+    function _withdraw(
+        uint256 _shares,
+        address _destination,
+        uint256
+    ) internal override returns (uint256) {
+        // Since ctoken's redeem function returns sucess codes
+        // we get the balance before and after minting to calculate amount
+        uint256 beforeBalance = token.balanceOf(address(this));
+
+        // Do the withdraw
+        uint256 redeemStatus = ctoken.redeem(_shares);
+        require(redeemStatus == 0, "compound redeem failed");
+
+        // Get underlying balance after withdrawing
+        uint256 afterBalance = token.balanceOf(address(this));
+        // Calculate the amount of funds that were freed
+        uint256 amountReceived = afterBalance - beforeBalance;
+        // Transfer the underlying to the destination
+        // 'token' is an immutable in WrappedPosition
+        token.transfer(_destination, amountReceived);
+
+        // Return the amount of underlying
+        return amountReceived;
+    }
+
+    /// @notice Get the underlying amount of tokens per shares given
+    /// @param _amount The amount of shares you want to know the value of
+    /// @return Value of shares in underlying token
+    function _underlying(uint256 _amount)
+        internal
+        override
+        view
+        returns (uint256)
+    {
+        // Load exchange rate
+        uint256 exchangeRate = ctoken.exchangeRateStored();
+
+        // Multiply _amount by exchange rate & correct for decimals
+        // We assume 8 decimals in the ctoken + 18 point decimal fix = 26
+        return ((_amount * exchangeRate) / (10**(26 - underlyingDecimals)));
+    }
+
+    /// @notice Collect the comp rewards accrued
+    /// @param _destination The address to send the rewards to
+    // TODO: add this back in when I figure out how to mock the owner funcitonality
+    function collectRewards(address _destination) external onlyAuthorized() {
+        // Set up input params for claimComp
+        address[] memory holder = new address[](1);
+        // Store contract address as an array
+        holder[0] = address(this);
+        CErc20Interface[] memory cTokens = new CErc20Interface[](1);
+        // Store cToken as an array
+        cTokens[0] = ctoken;
+
+        // claim the rewards
+        comptroller.claimComp(holder, cTokens, true, true);
+        // look up the comp balance to send
+        uint256 balance = comp.balanceOf(address(this));
+        // send to destination address
+        comp.transfer(_destination, balance);
```

Comment:
Are we keeping these comp rewards to us ?? Shouldn't we need to write mechanics where the user can claim their comp rewards on the basis of their shares?

Reply:
As discussed in our previous sync-ups, a separate contract will execute the collectRewards function